app/src/main/kotlin/com/mapbox/vision/teaser/replayer/ReplayModeFragment.kt

```diff
@@ -0,0 +1,188 @@
+    private fun onActivateMultiSelection() {
+        changeMultiSelection(true)
+        setMultiSelectionTitle()
+    }
+
+
+    private fun initDeleteSessionsButton() {
+        delete_sessions.setOnClickListener {
+            sessionsAdapter.let {
+                val sessionsPath = arguments?.getString(ARG_PARAM_SESSIONS_PATH)
+                if (it.isMultiSelection && sessionsPath != null) {
+                    for (session in it.selectedItems) {
+                        val sessionsFolder = File("$sessionsPath/$session")
+                        if (!sessionsFolder.deleteRecursively()) {
+                            Toast.makeText(requireContext(), "Error: Can't delete ${sessionsFolder.absoluteFile}", Toast.LENGTH_LONG).show()
+                        }
+                    }
+                    it.updateSessionsList()
+                    changeMultiSelection(false)
+                }
+            }
+        }
+    }
+
+    private fun onSessionClick(fileName: String) {
+        if (sessionsAdapter.isMultiSelection) {
+            setMultiSelectionTitle()
+        } else {
+            val listener = requireActivity()
+            if (listener is OnClickModeItemListener) {
+                listener.onClickSessionItem(fileName)
+                requireActivity().onBackPressed()
+            }
+        }
+    }
+
+    private fun onCameraClick() {
+        val listener = requireActivity()
+        if (listener is OnClickModeItemListener) {
+            listener.onClickCamera()
+            requireActivity().onBackPressed()
+        }
+    }
+
+    private fun setMultiSelectionTitle() {
+        sessionsAdapter.apply {
+            val count = getSelectedCount()
+            replay_fragment_title.text = requireContext().resources.getQuantityString(R.plurals.selected_items, count, count)
+        }
+    }
+
+    private fun changeMultiSelection(activate: Boolean) {
+        if (activate) {
+            sessionsAdapter.activateMultiSelection()
+            activateMultiSelectionVisibilityState(true)
+        } else {
+            sessionsAdapter.resetMultiSelection()
+            replay_fragment_title.setText(R.string.select_session_source)
+            activateMultiSelectionVisibilityState(false)
+        }
+    }
+
+    private fun activateMultiSelectionVisibilityState(visible: Boolean) {
+        val multiSelectionShow = if (visible) VISIBLE else GONE
+        val multiSelectionGone = if (visible) GONE else VISIBLE
+        edit_sessions_list.visibility = multiSelectionGone
+        select_all.visibility = multiSelectionShow
+        delete_sessions.visibility = multiSelectionShow
+        record_session.visibility = multiSelectionGone
+        back_button.visibility = multiSelectionGone
+        done_edit.visibility = multiSelectionShow
+    }
+
+    override fun onBackPressed(): Boolean {
+        if (sessionsAdapter.isMultiSelection) {
+            changeMultiSelection(false)
+            return false
+        }
+        return true
```

Comment:
It's a bit confusing that `true` signals that back is NOT processed in the fragment, and false means the opposite. WDYT about reverting semantics? 
Also pls add comment to the interface about semantics

Reply:
Okay, seems better, I'll change